sdk/cognitivelanguage/azure-ai-language-questionanswering/migration_guide.md

```diff
@@ -60,70 +69,381 @@ client = QuestionAnsweringClient(
-response = client.query_knowledge_base(
+from azure.ai.language.questionanswering import QuestionAnsweringClient
+from azure.ai.language.questionanswering.models import KnowledgeBaseAnswerContext
+
+client = QuestionAnsweringClient(endpoint=endpoint, credential=credential)
+
+response = client.get_answers(
     question="How long should my Surface battery last?",
     project_name="<my-qna-project-name>",
-    deployment_name="<my-qna-deployment-name>"
+    deployment_name="<my-qna-deployment-name>",
+    answer_context=KnowledgeBaseAnswerContext(
+        previous_qna_id=1
+    )
 )
+
 best_answers = [a for a in response.answers if a.confidence_score > 0.9]
 ```
 
+### Authoring Client
+
+#### Authenticating authoring client
+
+Previously in `azure-cognitiveservices-knowledge-qnamaker` you could create a `QnAMakerClient` by using `CognitiveServicesCredentials` from `msrest.authentication`:
+
+```python
+from msrest.authentication import CognitiveServicesCredentials
+from azure.cognitiveservices.knowledge.qnamaker import QnAMakerClient
+
+client = QnAMakerClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credentials=CognitiveServicesCredentials("API key")
+)
+```
+
+Now in `azure-ai-language-questionanswering` you can create a `QuestionAnsweringAuthoringClient` using an [AzureKeyCredential][azure_key_credential] or a token credential from the [azure-identity](https://pypi.org/project/azure-identity/) library:
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient
+
+client = QuestionAnsweringAuthoringClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credential=AzureKeyCredential("API key")
+)
+```
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient
+
+client = QuestionAnsweringAuthoringClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credential=DefaultAzureCredential()
+)
+```
+
+#### Creating knowledge base
+
+Previously in `azure-cognitiveservices-knowledge-qnamaker`, you could create a new knowledgebase using a `CreateKbDTO`:
+
+```python
+from msrest.authentication import CognitiveServicesCredentials
+from azure.cognitiveservices.knowledge.qnamaker.models import CreateKbDTO, QnADTO
+from azure.cognitiveservices.knowledge.qnamaker import QnAMakerClient
+
+client = QnAMakerClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credentials=CognitiveServicesCredentials("API key")
+)
+
+operation = client.knowledgebase.create(
+    create_kb_payload=CreateKbDTO(
+        name="<knowledgebase-name>",
+        qna_list=[
+            QnADTO(
+                questions=["questions"],
+                answer="answer"
+            )
+        ]
+        
+    )
+)
```

Comment:
Should there be more to this? You capture it in a variable called "operation", which to me seems like the thing from which you get a project. Conversely, in the new code you actually get the project back that you created.

Reply:
I've updated the code. For comparison, it doesn't seem you get a project back in track 1. I used "Operation" because that's what the model is called that is returned. This is an LRO in the track 1 library, but it's not actually represented as one in the generated code so the user has to check the status themselves. Ugly code but I'll include it so you can see how much better track 2 is. 😆